### PR TITLE
feat(composables): add onlyAvailable parameter

### LINKED ISSUE #1375: [BUG] Only show available payment and shipping methods
### Describe the bug

Currently, a user can choose between all payment and shipping methods. However, these can be blocked due to various reasons at runtime. In order to only fetch the list of currently valid payment or shipping methods, pass the `onlyAvailable=1` flag to the API endpoints `/shipping-method` and `/payment-method`.

### PR DESCRIPTION
**Pull request by mkucmus**
closes #1375 

### Checklist

- [x] the [list of features](docs/guide/FEATURELIST.md) is updated (if it's a feature and it's relevant)
- [x] I followed [contributing](https://github.com/DivanteLtd/shopware-pwa/blob/master/CONTRIBUTING.md) guidelines

### CODE REVIEW

File: packages/shopware-6-client/src/services/contextService.ts [hunk 1]
```diff
@@ -159,10 +159,14 @@ export async function getAvailableSalutations(
  * @beta
  */
 export async function getAvailablePaymentMethods(
-  contextInstance: ShopwareApiInstance = defaultInstance
+  contextInstance: ShopwareApiInstance = defaultInstance,
+  params: any = {}
```
Comment (patzick) [inline packages/shopware-6-client/src/services/contextService.ts:163]:
```suggestion
  params: { onlyAvailable?: boolean } = {}
```

File: packages/shopware-6-client/src/services/contextService.ts [hunk 2]
```diff
@@ -159,10 +159,14 @@ export async function getAvailableSalutations(
...
 ): Promise<PaymentMethod[]> {
   const resp = await contextInstance.invoke.get(
-    getContextPaymentMethodEndpoint()
+    getContextPaymentMethodEndpoint(),
+    {
+      params, // passing it as a last parameter won't cause a BC - to consider
```
Comment (patzick) [inline packages/shopware-6-client/src/services/contextService.ts:168]: is this comment necessary? There shouldn't be any BC here :)

File: packages/shopware-6-client/src/services/contextService.ts [hunk 3]
```diff
@@ -202,10 +206,14 @@ export async function setCurrentPaymentMethod(
  * @beta
  */
 export async function getAvailableShippingMethods(
-  contextInstance: ShopwareApiInstance = defaultInstance
+  contextInstance: ShopwareApiInstance = defaultInstance,
+  params: any = {} // passing it as a last parameter won't cause a BC - to consider
```
Comment (patzick) [inline packages/shopware-6-client/src/services/contextService.ts:210]:
```suggestion
  params: { onlyAvailable?: boolean } = {}
```

File: api/shopware-6-client.api.md [hunk 4]
```diff
@@ -165,13 +165,13 @@ export function getAvailableCurrencies(contextInstance?: ShopwareApiInstance): P
 export function getAvailableLanguages(contextInstance?: ShopwareApiInstance): Promise<Language[]>;
 
 // @beta (undocumented)
-export function getAvailablePaymentMethods(contextInstance?: ShopwareApiInstance): Promise<PaymentMethod[]>;
+export function getAvailablePaymentMethods(contextInstance?: ShopwareApiInstance, params?: any): Promise<PaymentMethod[]>;
```
Comment (patzick) [inline api/shopware-6-client.api.md:168]: types should change here, after local rebuild

**Review by patzick — CHANGES_REQUESTED** [review]
Looks good, we should add more specific typing though :)

File: packages/shopware-6-client/src/services/contextService.ts (hunk 2 above)
Comment (mkucmus) [reply packages/shopware-6-client/src/services/contextService.ts]: yes, removed

File: packages/shopware-6-client/src/services/contextService.ts (hunk 3 above)
Comment (mkucmus) [reply packages/shopware-6-client/src/services/contextService.ts]: cool, thanks! added a more detailed type

File: packages/shopware-6-client/src/services/contextService.ts (hunk 1 above)
Comment (mkucmus) [reply packages/shopware-6-client/src/services/contextService.ts]: added!

**Review by patzick — APPROVED** [review]
👌  excellent

**Pull request merged**